client-app/shared/files/utils/index.ts

```diff
@@ -0,0 +1,29 @@
+import { useAuth } from "@/core/composables";
+
+const { headers } = useAuth();
+export async function generateLinkAndDownloadFile(fileUrl: string, fileName: string) {
+  try {
+    const response = await fetch(fileUrl, {
```

Comment:
use client-app/core/api/common/composables/useFetch.ts

there is interceptor inside which add the header